components/dashboards-web-component/src/designer/DashboardDesigner.jsx

```diff
@@ -265,11 +304,82 @@ export default class DashboardDesigner extends Component {
                 pageId = this.state.dashboard.landingPage;
             }
 
-            let dashboard = this.state.dashboard;
+            let dashboard = this.getDashboard();
+
+            let optionsMap = {};
+
+            /**
+             * helper function to get the page from a given dashboard given the pageId
+             * */
+            function getPage(dashboard, pageID) {
+                let pages = dashboard.pages;
+                if (pages) {
+                    for (let i = 0; i < pages.length; i++) {
+                        let page = pages[i];
+                        if (page.id === pageID) {
+                            return page;
+                        }
+                    }
+                }
+                return null;
+            }
+
+            let page = getPage(dashboard, pageId);
+
+            /**
+             * helper function to traverse the content of a page and find widgets to extract their options to a
+             * given map
+             * @param page : the page to be traversed
+             * @param OptionsMap : the map to extract option configurations
+             * */
+            function searchForWidgetsInAPage(page, OptionsMap) {
+                let obj = page;
+                if (obj.type && obj.type === "component" && obj.props && obj.props.id) {
```

Comment:
Please write it like the following (Note single quotes for 'component' too)

`if (obj.type && obj.props && obj.type === 'component' && obj.props.id) {`